src/mlpack/methods/edge_boxes/feature_extraction.hpp

```diff
@@ -0,0 +1,262 @@
+/**
+ * @file feature_extraction.hpp
+ * @author Nilay Jain 
+ *
+ * Feature Extraction for the edge_boxes algorithm.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_HPP
+#include <mlpack/core.hpp>
+#include "feature_parameters.hpp"
+namespace mlpack {
+namespace structured_tree {
+
+template <typename MatType = arma::mat, typename CubeType = arma::cube>
+class StructuredForests
+{
+ private:
+  FeatureParameters params;
+
+ public:
+
+  /**
+   * Constructor: stores all the parameters in an object
+   * of feature_parameters class.
+   * @param F FeatureParameters object which stores necessary parameters.
+   */ 
+  StructuredForests(FeatureParameters F);
+
+  void PrepareData(const MatType& Images, const MatType& Boundaries,
+            const MatType& Segmentations);
+
+  /**
+   * Get Dimensions of Features
+   * @param FtrDim Output vector that contains the result dimensions. 
+   */
+  void GetFeatureDimension(arma::vec& FtrDim);
+  
+  /**
+   * Computes distance transform of 1D vector f.
+   * @param f input vector whose distance transform is to be found.
+   * @param n size of the Output vector to be made.
+   * @param inf a large double value.
+   * @param d Output vector which stores distance transform of f.
+   */
+  void DistanceTransform1D(const arma::vec& f, const size_t n,
+                           const double inf, arma::vec& d);
+  
+  /**
+   * Computes distance transform of a 2D array
+   * @param Im input array whose distance transform is to be found.
+   * @param inf a large double value.
+   */
+  void DistanceTransform2D(MatType &Im, const double inf);
```

Comment:
Note that Variable, reference, and pointer names begin with a lower case letter. Take a look at: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more informations.